src/vos/vos_obj.c

```diff
@@ -499,6 +502,10 @@ vos_obj_punch(daos_handle_t coh, daos_unit_oid_t oid, daos_epoch_t epoch,
 					obj->obj_df->vo_max_write = epr.epr_hi;
 			}
 
+			if (rc == 0)
+				rc = vos_obj_flags_set(vos_cont2umm(cont), &obj->obj_df->vo_tree,
+						       &cont->vc_cont_df->cd_obj_root);
```

Comment:
Why container isn't marked?

Reply:
You mean pool?   Container is marked by setting bits in the object tree root.

Reply:
Yeah, I meant container, I was thinking vos_obj_flags_set() only update flags for object.